app/scripts/modules/netflix/ci/services/build.read.service.spec.ts

```diff
@@ -0,0 +1,136 @@
+import {SETTINGS} from 'core/config/settings';
+import {Api, API_SERVICE} from 'core/api/api.service';
+import {
+  CI_BUILD_READ_SERVICE, CiBuildReader, ICiBuild,
+  ICiBuildOutputConfig
+} from 'netflix/ci/services/build.read.service';
+import {CI_FILTER_MODEL, CiFilterModel} from 'netflix/ci/ci.filter.model';
+
+describe('CiBuildReader', () => {
+
+  const CI_BUILD_URL = `${SETTINGS.gateUrl}/ci/builds`;
+  let $http: IHttpBackendService;
+  let API: Api;
+  let ciBuildReader: CiBuildReader;
+  let ciFilterModel: CiFilterModel;
+
+  beforeEach(mock.module(API_SERVICE, CI_BUILD_READ_SERVICE, CI_FILTER_MODEL));
+  beforeEach(mock.inject((_$httpBackend_: IHttpBackendService,
+                          _API_: Api,
+                          _ciBuildReader_: CiBuildReader,
+                          _CiFilterModel_: CiFilterModel) => {
+    $http = _$httpBackend_;
+    API = _API_;
+    ciBuildReader = _ciBuildReader_;
+    ciFilterModel = _CiFilterModel_;
+  }));
+
+  beforeEach(() => {
+    ciFilterModel.searchFilter = 'none';
+  });
+
+  afterEach(() => {
+    ciFilterModel.searchFilter = null;
+  });
+
+  afterEach(function () {
+    $http.verifyNoOutstandingExpectation();
+    $http.verifyNoOutstandingRequest();
+  });
+
+  function getBuild(buildId = 'buildId', completionStatus = 'COMPLETE'): ICiBuild {
+    return {
+      buildNumber: Math.floor(Math.random() * 100),
+      change: null,
+      completedAt: moment.now(),
+      completionStatus: completionStatus,
+      id: buildId,
+      repositoryId: 'repoId',
+      startedAt: moment.now() - 10000
+    };
+  }
+
+  it('should retrieve a list of transformed builds', () => {
+    const repoType = 'r_rt';
+    const projectKey = 'r_pk';
+    const repoSlug = 'r_rs';
+    $http.expectGET(`${CI_BUILD_URL}?repoType=${repoType}&projectKey=${projectKey}&repoSlug=${repoSlug}&filter=${ciFilterModel.searchFilter}`)
+      .respond(200, {data: [getBuild(), getBuild('buildId', 'INCOMPLETE')]});
+
+    let result: ICiBuild[] = null;
+    ciBuildReader.getBuilds(repoType, projectKey, repoSlug).then((builds: ICiBuild[]) => result = builds);
+    $http.flush();
+    expect(result.length).toBe(2);
+    result.forEach((build: ICiBuild) => {
+      expect(build.startTime).toBe(build.startedAt);
+      expect(build.endTime).toBe(build.completedAt);
+      expect(build.isRunning).toBe(build.completionStatus === 'INCOMPLETE');
+    })
+  });
+
+  it('should retrieve a list of running builds in a promise', () => {
+    const repoType = 'r_rt';
+    const projectKey = 'r_pk';
+    const repoSlug = 'r_rs';
+    $http.expectGET(`${CI_BUILD_URL}?repoType=${repoType}&projectKey=${projectKey}&repoSlug=${repoSlug}&completionStatus=INCOMPLETE`)
+      .respond(200, []);
+
+    const result: IPromise<ICiBuild[]> = ciBuildReader.getRunningBuilds(repoType, projectKey, repoSlug);
+    $http.flush();
+    expect(result).toBePromiseLike();
+  });
+
+  it('should retrieve the transformed build details in a promise', () => {
```

Comment:
Not sure this name is accurate. Either `should retrieve the transformed build details` or `should return the transformed build details in promise`.